stock_production_lot_available_at_date/report/report_stock_production_lot_at_date.py

```diff
@@ -0,0 +1,69 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import api, models, fields
+
+
+class ReportStockProductionLotDate(models.TransientModel):
+    _name = "report.stock.production.lot.date"
+    _description = "Stock Production Lot Date Report"
+
+    lot_id = fields.Many2one('stock.production.lot')
+    product_id = fields.Many2one('product.template')

```

Comment:
the right product type for this report should be the product variant, not the template IMO, because lot and other models (like stock moves) are actually linked to `product.product`.